src/Recorder/FilesystemRecorder.php

```diff
@@ -59,15 +65,19 @@ public function replay(string $name): ?ResponseInterface
 
         $this->log('Response replayed from {filename}', $context);
 
-        return Psr7\parse_response(file_get_contents($filename));
+        return Message::parseResponse(file_get_contents($filename));
     }
 
     public function record(string $name, ResponseInterface $response): void
     {
         $filename = "{$this->directory}$name.txt";
         $context = compact('name', 'filename');
 
-        $this->filesystem->dumpFile($filename, Psr7\str($response));
+        $content = Message::toString($response);
+        foreach ($this->filters as $regularExpression => $replacement) {
+            $content = preg_replace($regularExpression, $replacement, $content);
```

Comment:
`preg_replace` supports array for regex & replacement, which could avoid the need for a loop and probably increase performance.

Reply:
Didn't know that. Thanks for the advice. it's done